quantus/metrics/base.py

```diff
@@ -1,105 +1,331 @@
 """This module implements the base class for creating evaluation measures."""
 import warnings
-import numpy as np
-from typing import Union, List, Dict, Any
+from abc import abstractmethod
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
 import matplotlib.pyplot as plt
-from ..helpers.utils import *
-from ..helpers.asserts import *
-from ..helpers.plotting import *
-from ..helpers.normalise_func import *
-from ..helpers.warn_func import *
+import numpy as np
+from tqdm import tqdm
+
+from ..helpers import utils
+from ..helpers import asserts
+from ..helpers.model_interface import ModelInterface
+from ..helpers.normalise_func import normalise_batch, normalise_by_negative
+from ..helpers import warn_func
 
 
 class Metric:
     """
     Implementation base Metric class.
     """
 
-    @attributes_check
-    def __init__(self, *args, **kwargs):
+    @asserts.attributes_check
+    def __init__(
+            self,
+            abs: bool = False,
+            normalise: bool = False,
+            normalise_func: Optional[Callable] = None,
+            normalise_func_kwargs: Optional[Dict[str, Any]] = None,
+            softmax: bool = False,
```

Comment:
Can't we have something called predict_kwargs? or model_kwargs?

Reply:
I put it in `__call__`, as I didn't think that it makes sense to have model kwargs before passing the model (which is passed just in `__call__`, not in `__init__`.